chain-core/src/state/account/op/data/attribute.rs

```diff
@@ -0,0 +1,42 @@
+use parity_scale_codec::{Decode, Encode, Error, Input, Output};
+#[cfg(not(feature = "mesalock_sgx"))]
+use serde::{Deserialize, Serialize};
+
+/// attributes in StakedState-related transactions
+#[derive(Debug, Default, PartialEq, Eq, Clone)]
+#[cfg_attr(not(feature = "mesalock_sgx"), derive(Serialize, Deserialize))]
+pub struct StakedStateOpAttributes {
+    pub chain_hex_id: u8,
+    pub app_version: u64,
+}
+
+impl StakedStateOpAttributes {
+    pub fn new(chain_hex_id: u8) -> Self {
+        StakedStateOpAttributes {
+            chain_hex_id,
+            app_version: crate::APP_VERSION,
+        }
+    }
+}
+
+impl Encode for StakedStateOpAttributes {
+    fn encode_to<EncOut: Output>(&self, dest: &mut EncOut) {
+        dest.push_byte(self.chain_hex_id);
```

Comment:
Does this struct need to be prepared for future extension?

Reply:
can't think of any immediate extension -- same for the transfer attributes -- but here it may change, so could be tagged